src/gui/qgsmapcanvas.cpp

```diff
@@ -778,8 +796,26 @@ QgsRectangle QgsMapCanvas::imageRect( const QImage& img, const QgsMapSettings& m
 
 void QgsMapCanvas::mapUpdateTimeout()
 {
-  const QImage& img = mJob->renderedImage();
-  mMap->setContent( img, imageRect( img, mJob->mapSettings() ) );
+  QImage img = mJob->renderedImage();
+  magnifyImage( img );
+
+  mMap->setContent( img, imageRect( img, mSettings ) );
+}
+
+void QgsMapCanvas::magnifyImage( QImage &img )
+{
+  if ( mMagnificationLevel != 1.0 )
+  {
+    img = img.scaled( img.width() * mMagnificationLevel,
```

Comment:
@nyalldawson
 
 Yes I tried that and I agree that everything is correctly scaled without any pixelation.
 
 But there's a side effect that we don't want here (and it's because of that side effect that I
 asked some questions above). That's for example what I have when I play with the map settings
 DPI :
 
![sc](https://cloud.githubusercontent.com/assets/9266424/14743974/c3f834b6-08a4-11e6-9eca-eb61a2774d91.png)

 
 In our usecase, we want a magnification tool which provides the exact same rendering
 whatever the level of magnification (except the extent of course). But playing with DPI has a direct consequency on labels' position.
 
 This side effect is due to the fact that the scaleFactor of QgsRenderContext depends on the DPI set in QgsMapSettings.
 
 Thank you for your involvement :)